knowledge-base/upload-validateantiforgerytoken.md

```diff
@@ -0,0 +1,226 @@
+    private string? AntiforgeryFormFieldName { get; set; }
+    private string? AntiforgeryFormValue { get; set; }
+
+    private void OnUploadUpload(UploadEventArgs args)
+    {
+        // There is no need to post both antiforgery header and data.
+        // Only one of them is enough.
+
+        args.RequestHeaders.Add(AntiforgeryHeaderName, AntiforgeryHeaderToken);
+        args.RequestData.Add(AntiforgeryFormFieldName, AntiforgeryFormValue);
+    }
+
+    private void OnUploadRemove(UploadEventArgs args)
+    {
+        // There is no need to post both antiforgery header and data.
+        // Only one of them is enough.
+
+        args.RequestHeaders.Add(AntiforgeryHeaderName, AntiforgeryHeaderToken);
+        args.RequestData.Add(AntiforgeryFormFieldName, AntiforgeryFormValue);
+    }
+
+    protected override void OnInitialized()
+    {
+        // Obtain the antiforgery header name and value.
+        if (HttpContextAccessor.HttpContext != null)
+        {
+            var afTokenSet = Antiforgery.GetAndStoreTokens(HttpContextAccessor.HttpContext);
+            AntiforgeryHeaderName = afTokenSet.HeaderName;
+            AntiforgeryHeaderToken = afTokenSet.RequestToken;
+        }
+
+        // Obtain the antiforgery form field name and value.
+        var afRequestToken = AfStateProvider.GetAntiforgeryToken();
+        if (afRequestToken != null)
+        {
+            AntiforgeryFormFieldName = afRequestToken.FormFieldName;
+            AntiforgeryFormValue = afRequestToken.Value;
+        }
+
+        base.OnInitialized();
+    }
+
+    private string ToAbsoluteUrl(string url)
+    {
+        return $"{NavigationManager.BaseUri}{url}";
+    }
+}
+````
+````Controller
+using Microsoft.AspNetCore.Mvc;
+
+namespace BlazorAppName.Controllers
+{
+    [ValidateAntiForgeryToken]
+    [Route("api/[controller]/[action]")]
+    public class UploadController : ControllerBase
+    {
+        public IWebHostEnvironment HostingEnvironment { get; set; }
+
+        public UploadController(IWebHostEnvironment hostingEnvironment)
+        {
+            HostingEnvironment = hostingEnvironment;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Save(IFormFile files)
+        {
+            // Save the file...
+
+            return new EmptyResult();
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Remove([FromForm] string files)
+        {
+            // Delete the file...
+
+            return new EmptyResult();
+        }
+    }
+}
+````
+
+> This article contains code snippets and suggestions that relate to general .NET programming and antiforgery setup of a Blazor application. The provided implementation is just an example and is strictly outside the Telerik support scope. The primary resource for antiforgery configuration is the Microsoft documentation. See [Blazor authentication and authorization](https://learn.microsoft.com/en-us/aspnet/core/blazor/security/).
```

Comment:
Personally, I would place this disclaimer at the top of the page or at the beginning of the Solution section.

Reply:
This makes sense from one point of view, but the disclaimer won't look good there, because it's dismissive (as it should be).